class-components/src/app/lib/schema.ts

```diff
@@ -0,0 +1,113 @@
+import { z } from 'zod';
+
+const MAX_FILE_SIZE = 5 * 1024 * 1024;
+const ACCEPTED_IMAGE_TYPES = ['image/jpeg', 'image/png'];
+
+export const formSchema = z
+  .object({
+    name: z
+      .string()
+      .trim()
+      .min(1, { message: 'Name is required.' })
+      .refine(
+        (val) => {
+          if (val.length === 0) return true;
+          return /^[A-Z]/.test(val);
+        },
+        {
+          message: 'Name must start with a capital letter.',
+        }
+      ),
```

Comment:
a bit difficult to read. 

Maybe smth like 
```suggestion
  .object({
    name:  z
  .string()
  .trim()
  .min(1, { message: 'Name is required.' })
  .refine(
    (val) => /^[A-Z]/.test(val),
    { message: 'Name must start with a capital letter.' }
  );
      ),
```

Reply:
same for other, to remove the check for val.length === 0